uritemplate/validator.py

```diff
@@ -0,0 +1,58 @@
+"""Module containing all the validation logic for uritemplate."""
+from uritemplate import exceptions as exc
+
+
+class Validator(object):
+    """Provide configurable validation of URITemplate objects.
+
+    .. versionadded:: 3.1.0
+
+    .. code-block::
+
+        from uritemplate import URITemplate, Validator
+
+
+    """
+
+    def __init__(self):
+        """Initialize our validator."""
+        self.enforcing_unbalanced_braces = True
+
+    def allow_unbalanced_braces(self):
+        """Allow a template to have unbalanced braces.
+
+        .. versionadded:: 3.1.0
+
+        Returns the validator instance.
+        """
+        self.enforcing_unbalanced_braces = False
+        return self
+
+    def force_balanced_braces(self):
+        """Force a template to have balanced braces.
+
+        .. versionadded:: 3.1.0
+
+        Returns the validator instance.
+        """
+        self.enforcing_unbalanced_braces = True
+        return self
+
+    def validate(self, template):
+        """Validate that a template meets the parameters.
+
+        .. versionadded:: 3.1.0
+
+        :raises: uritemplate.exceptions.InvalidTemplate
+        :raises: uritemplate.exceptions.UnbalancedBraces
+        """
+        if self.enforcing_unbalanced_braces:
```

Comment:
> (about whether someone should want to write if somevalidator.enforcing_unbalanced_braces in external code bases)

I don't think it would hurt if someone wanted to print out the configuration in a log message for themselves, especially if their validation logic is somewhat dynamic.